### ISSUE TITLE
Large slowdown from SyncBatchNorm

### ISSUE THREAD

**Issue by austinmw** [horovod/horovod#2450]
**Environment:**
1. Framework: PyTorch
2. Framework version: 1.6.0
3. Horovod version: 0.20.3
4. MPI version: 4.0.5
5. CUDA version: 11.0
6. NCCL version: 2.7.8
7. Python version: 3.6
8. OS and version: Ubuntu 16.04

**Your question:**

I'm currently working on an AWS ml.p3.8xlarge instance type (4x NVIDIA V100). I'm working with a HRNet-18 model.

I recently updated my training code to use Horovod instead of DataParallel, and am currently speed testing single node 4x GPU. 

I noticed that when I replace the `torch.nn.BatchNorm2d` layers with `horovod.torch.sync_batch_norm.SyncBatchNorm` layers, my network forward pass time increases by ~4x (approximately 1.2s up to 4.5s). 

Are there additional configuration options that should be added, or is SyncBatchNorm necessarily much slower?

**Comment by tgaddair** [comment]
Hey @austinmw, `SyncBatchNorm` is definitely slower, but shouldn't typically be that much slower.  One thing we often see is that tuning the cycle time (`HOROVOD_CYCLE_TIME`) or fusion buffer threshold (`HOROVOD_FUSION_THRESHOLD`) can have a significant effect on improving sync batch norm performance.  You may also want to try enabling fp16 compression to see if that has an effect.

**Comment by austinmw** [comment]
Thanks!

**Comment by edumotya** [comment]
I'm also encountering very large slowdowns while utilizing synchronized batch normalization, in my case with tensorflow. I've configured it for FP16 compression and made adjustments to cycle time and fusion buffer threshold using Horovod's autotuning feature, but unfortunately, these changes haven't resulted in any noticeable performance improvement.

torus_allreduce | cache_enabled | cycle_time_ms | tensor_fusion_threshold | score
-- | -- | -- | -- | --
0 | 0 | 5 | 4 | 0.0137566
0 | 0 | 50 | 32 | 0.0146798
0 | 0 | 25 | 16 | 0.0145344
0 | 0 | 10 | 8 | 0.0137237
0 | 0 | 25.0098 | 15.9936 | 0.0140181
0 | 0 | 79.3112 | 17.359 | 0.0137522
0 | 0 | 61.9099 | 10.9202 | 0.0137085
0 | 0 | 43.7195 | 36.8173 | 0.0137564
0 | 0 | 74.8363 | 63.6205 | 0.0141782
0 | 0 | 44.4391 | 10.778 | 0.0144531
0 | 0 | 29.1699 | 8.95695 | 0.0147803
0 | 0 | 70.3313 | 25.5323 | 0.0139509
0 | 0 | 56.6932 | 23.2429 | 0.0139497
0 | 0 | 89.8817 | 5.29874 | 0.0141698
0 | 0 | 72.6778 | 62.2072 | 0.0143431
0 | 0 | 1.61695 | 61.7163 | 0.0144063
0 | 0 | 96.3817 | 62.7478 | 0.0139072
0 | 0 | 1.17246 | 29.1327 | 0.0141071
